dotCMS/src/main/java/com/dotcms/rendering/velocity/viewtools/BrowserAPI.java

```diff
@@ -254,256 +260,121 @@ public WfData(Contentlet contentlet, List<Integer> permissions, User user, boole
-				    
-					Map<String, Object> pageMap = page.getMap();
-					pageMap.put("mimeType", "application/dotpage");
-					pageMap.put("permissions", permissions);
-					pageMap.put("name", page.getPageUrl());
-					pageMap.put("description", page.getFriendlyName());
-					pageMap.put("extension", "page");
-					pageMap.put("isContentlet", isContentlet);
-					
-					if(wfdata!=null ) {
-    		            pageMap.put("wfActionMapList", wfdata.wfActionMapList);
-    	                pageMap.put("contentEditable", wfdata.contentEditable);
-					}
-					
-					if(isContentlet) {
-					    pageMap.put("identifier", page.getIdentifier());
-		                pageMap.put("inode", page.getInode());
-					    pageMap.put("languageId", ((Contentlet)page).getLanguageId());
-					    
-					    Language lang = APILocator.getLanguageAPI().getLanguage(((Contentlet)page).getLanguageId());
-					    
-					    pageMap.put("languageCode", lang.getLanguageCode());
-		                pageMap.put("countryCode", lang.getCountryCode());
-		                pageMap.put("isLocked", page.isLocked());
-		                pageMap.put("languageFlag", LanguageUtil.getLiteralLocale(lang.getLanguageCode(), lang.getCountryCode()));
-		            
-		              
-					}
-
-                    pageMap.put("hasLiveVersion", APILocator.getVersionableAPI().hasLiveVersion(page));
-					pageMap.put("statusIcons", UtilHTML.getStatusIcons(page));
-					pageMap.put("hasTitleImage",String.valueOf(((Contentlet)page).getTitleImage().isPresent()));
-					pageMap.put("__icon__", "pageIcon");
-					returnList.add(pageMap);
-				}
-			}
+            if (parent != null) {
+                if (showWorking) {
+                    links.addAll(folderAPI.getLinks(parent, true, false, user, false));
+                } else {
+                    links.addAll(folderAPI.getLiveLinks(parent, user, false));
+                }
+                if (showArchived)
+                    links.addAll(folderAPI.getLinks(parent, true, showArchived, user, false));
+            } else {
+                links = folderAPI.getLinks(host, true, showArchived, user, false);
+            }
 
-		}
 
-		List<Versionable> files = new ArrayList<Versionable>();
-		try {
+	        for (Link link : links) {
+
+	            List<Integer> permissions2 = permissionAPI.getPermissionIdsFromRoles(link, roles, user);
+
+	            if (permissions2.contains(PERMISSION_READ)) {
+	                Map<String, Object> linkMap = link.getMap();
+	                linkMap.put("permissions", permissions2);
+	                linkMap.put("mimeType", "application/dotlink");
+	                linkMap.put("name", link.getTitle());
+	                linkMap.put("description", link.getFriendlyName());
+	                linkMap.put("extension", "link");
+	                linkMap.put("hasLiveVersion", APILocator.getVersionableAPI().hasLiveVersion(link));
+	                linkMap.put("statusIcons", UtilHTML.getStatusIcons(link));
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The String literal "statusIcons" appears 4 times in this file; the first occurrence is on line 291](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337824)